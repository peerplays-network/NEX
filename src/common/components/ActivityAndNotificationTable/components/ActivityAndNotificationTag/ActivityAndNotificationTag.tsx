import counterpart from "counterpart";

import { Tag } from "../../../../../ui/src";

type Props = {
  type: string;
};

export const ActivityAndNotificationTag = ({ type }: Props): JSX.Element => {
  return (
    <Tag key={type}>
      {counterpart.translate(`transaction.trxTypes.${type}.title`)}
    </Tag>
  );
};
