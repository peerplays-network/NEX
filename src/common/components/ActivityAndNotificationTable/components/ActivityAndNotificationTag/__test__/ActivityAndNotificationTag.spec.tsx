import { render } from "@testing-library/react";
import React from "react";

import { ActivityAndNotificationTag } from "../ActivityAndNotificationTag";

describe("Activity Tag", () => {
  it("renders correctly for account creation", () => {
    const dummyComponentProps = {
      type: "account_create",
    };
    const { asFragment } = render(
      <ActivityAndNotificationTag {...dummyComponentProps} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly for account update", () => {
    const dummyComponentProps = {
      type: "account_update",
    };
    const { asFragment } = render(
      <ActivityAndNotificationTag {...dummyComponentProps} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
