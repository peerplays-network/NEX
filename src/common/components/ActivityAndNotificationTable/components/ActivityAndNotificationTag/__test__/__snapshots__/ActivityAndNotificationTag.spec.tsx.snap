// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Tag renders correctly for account creation 1`] = `
<DocumentFragment>
  <span
    class="Tag-sc-5420511e-0 JuGDv"
  >
    Create account
  </span>
</DocumentFragment>
`;

exports[`Activity Tag renders correctly for account update 1`] = `
<DocumentFragment>
  <span
    class="Tag-sc-5420511e-0 JuGDv"
  >
    Update account
  </span>
</DocumentFragment>
`;
